src/k8s-engine/lima.ts

```diff
@@ -241,6 +237,21 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     return this.#desiredPort;
   }
 
+  protected async ensureArchitectureMatch() {
+    if (os.platform().startsWith('darwin')) {
+      const fileArch = this.arch === 'aarch64' ? 'arm64' : this.arch;
+      const { stdout } = await childProcess.spawnFile(
+        'file', [this.limactl],
+        { stdio: ['inherit', 'pipe', console] });
+
+      if (!stdout.trim().match(`executable ${ fileArch }$`)) {
+        const otherArch = { aarch64: 'x86_64', x86_64: 'aarch64' }[this.arch];
+
+        throw new Error(`Rancher Desktop for ${ otherArch } does not work on ${ this.arch }.`);
```

Comment:
Please throw a [`KubernetesError`](https://github.com/rancher-sandbox/rancher-desktop/blob/abdb26fee17fb92327d19dab9200e751fdea0cc1/src/k8s-engine/k8s.ts#L19); we have [special handling](https://github.com/rancher-sandbox/rancher-desktop/blob/abdb26fee17fb92327d19dab9200e751fdea0cc1/background.ts#L591) to drop the `unknown error` part of the message (because it's known).

Even better, delete `LimaSudoRejectionError` and make a flag on `KuberentesError` to control quitting the app.